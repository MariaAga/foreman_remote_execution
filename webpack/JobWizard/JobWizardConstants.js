import { foremanUrl } from 'foremanReact/common/helpers';

export const JOB_TEMPLATES = 'JOB_TEMPLATES';
export const JOB_CATEGORIES = 'JOB_CATEGORIES';
export const JOB_TEMPLATE = 'JOB_TEMPLATE';
export const templatesUrl = foremanUrl('/api/v2/job_templates');
