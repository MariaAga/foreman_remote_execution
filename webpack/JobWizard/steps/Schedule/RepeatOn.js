import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  TextInput,
  Grid,
  GridItem,
  FormGroup,
  Checkbox,
} from '@patternfly/react-core';
import { translate as __, documentLocale } from 'foremanReact/common/I18n';
import { SelectField } from '../form/SelectField';
import { repeatTypes } from '../../JobWizardConstants';

const getWeekDays = () => {
  const locale = documentLocale().replace(/-/g, '_');
  const baseDate = new Date(Date.UTC(2017, 0, 2)); // just a Monday
  const weekDays = [];
  for (let i = 0; i < 7; i++) {
    try {
      weekDays.push(baseDate.toLocaleDateString(locale, { weekday: 'short' }));
    } catch {
      weekDays.push(baseDate.toLocaleDateString('en', { weekday: 'short' }));
    }
    baseDate.setDate(baseDate.getDate() + 1);
  }
  return weekDays;
};

export const RepeatOn = ({
  repeatType,
  setRepeatType,
  repeatAmount,
  setRepeatAmount,
}) => {
  const [repeatValidated, setRepeatValidated] = useState('default');
  const handleRepeatInputChange = newValue => {
    setRepeatValidated(newValue >= 1 ? 'default' : 'error');
    setRepeatAmount(newValue);
  };
  const days = getWeekDays();
  const [repeatOn, setRepeatOn] = useState([]);
  const handleChangeDays = (checked, event) => {
    const { target } = event;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const { name } = target;
    setRepeatOn(oldValue => ({ ...oldValue, [name]: value }));
  };
  return (
    <Grid id="repeat-on">
      <GridItem span={6}>
        <SelectField
          fieldId="repeat-select"
          options={Object.values(repeatTypes)}
          setValue={newValue => {
            setRepeatType(newValue);
            if (newValue === repeatTypes.noRepeat) {
              setRepeatValidated('default');
            }
          }}
          value={repeatType}
        />
      </GridItem>
      <GridItem span={1} />
      <GridItem span={5}>
        <FormGroup
          isInline
          helperTextInvalid={__('Repeat amount can only be a positive number')}
          validated={repeatValidated}
        >
          {repeatType === repeatTypes.weekly ? (
            <div id="repeat-on-grid">
              {days.map((day, index) => (
                <Checkbox
                  key={index}
                  isChecked={repeatOn[index]}
                  name={index}
                  id={`repeat-on-day-${index}`}
                  onChange={handleChangeDays}
                  label={day}
                />
              ))}
            </div>
          ) : (
            <TextInput
              isDisabled={repeatType === repeatTypes.noRepeat}
              id="repeat-amount"
              value={repeatAmount}
              type="text"
              onChange={newValue => handleRepeatInputChange(newValue)}
              // placeholder={__('Repeat N times')}
            />
          )}
        </FormGroup>
      </GridItem>
    </Grid>
  );
};

RepeatOn.propTypes = {
  repeatType: PropTypes.oneOf(Object.values(repeatTypes)).isRequired,
  setRepeatType: PropTypes.func.isRequired,
  repeatAmount: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  setRepeatAmount: PropTypes.func.isRequired,
};
