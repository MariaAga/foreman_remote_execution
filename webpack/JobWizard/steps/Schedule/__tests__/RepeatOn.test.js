import React from 'react';
import * as patternfly from '@patternfly/react-core';
import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { RepeatOn } from '../RepeatOn';
import { repeatTypes } from '../../../JobWizardConstants';

jest.spyOn(patternfly, 'FormGroup');
patternfly.FormGroup.mockImplementation(props => <div props={props} />);

const fixtures = {
  renders: {
    repeatType: Object.values(repeatTypes)[1],
    setRepeatType: jest.fn(),
    repeatAmount: 1,
    setRepeatAmount: jest.fn(),
  },
};

describe('RepeatOn', () => {
  testComponentSnapshotsWithFixtures(RepeatOn, fixtures);
});
