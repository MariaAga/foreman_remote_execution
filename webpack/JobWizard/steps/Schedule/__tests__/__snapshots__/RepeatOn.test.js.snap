// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepeatOn renders 1`] = `
<Grid>
  <GridItem
    span={6}
  >
    <SelectField
      fieldId="repeat-select"
      label={null}
      options={
        Array [
          "Does not repeat",
          "Cronline",
          "Monthly",
          "Weekly",
          "Daily",
          "Hourly",
        ]
      }
      setValue={[Function]}
      value="Cronline"
    />
  </GridItem>
  <GridItem
    span={1}
  />
  <GridItem
    span={5}
  >
    <mockConstructor
      helperTextInvalid="Repeat amount can only be a positive number"
      isInline={true}
      validated="default"
    >
      <TextInput
        id="repeat-amount"
        isDisabled={false}
        onChange={[Function]}
        placeholder="Repeat N times"
        type="text"
        value={1}
      />
    </mockConstructor>
  </GridItem>
</Grid>
`;
